### PR TITLE
Improve code quality

### LINKED ISSUE #54: Improve code quality & use some C#11 features
We should improve the overall code quality by eliminating some issues on Codacy. In addition, we should make use of some of the new features introduced in C#11 like [Required Members](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required), which can simplify the code.

### PR DESCRIPTION
**Pull request by snixtho**
resolves #54

### CODE REVIEW

File: src/EvoSC.Common/Config/ConfigServiceExtensions.cs [hunk 1]
```diff
@@ -11,6 +11,12 @@ public static class ConfigServiceExtensions
 {
     private const string MainConfigFile = "config/main.toml";
     
+    /// <summary>
+    /// Set up and parse EvoSC's base configuration. This method also adds the configuration
+    /// to the service container.
+    /// </summary>
+    /// <param name="services"></param>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Config/ConfigServiceExtensions.cs:18]: Could use a quick and short description of services.

File: src/EvoSC.Common/Config/Models/IDatabaseConfig.cs [hunk 2]
```diff
@@ -6,6 +6,9 @@ namespace EvoSC.Common.Config.Models;
 
 public interface IDatabaseConfig
 {
+    /// <summary>
+    /// Database types supported by the application.
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Config/Models/IDatabaseConfig.cs:10]: Database types supported by the application e.g MySQL, PostgreSQL.

File: src/EvoSC.Common/Config/Stores/StoreExtensions.cs [hunk 3]
```diff
@@ -4,6 +4,13 @@ namespace EvoSC.Common.Config.Stores;
 
 public static class StoreExtensions
 {
+    /// <summary>
+    /// Use the TOML file store for this config.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="path">The path to the config file.</param>
+    /// <typeparam name="TInterface"></typeparam>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Config/Stores/StoreExtensions.cs:12]: Could use a description of what the TInterface is

File: src/EvoSC.Common/Events/Attributes/SubscribeAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        Priority = priority;
+        IsAsync = isAsync;
+    }
+
+    /// <summary>
+    /// Create a new event subscription.
+    /// </summary>
+    /// <param name="name">Must be an enumeration. Name of the event.</param>
+    public SubscribeAttribute(object name) : this((name as Enum).GetEventIdentifier()) {}
+    /// <summary>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Events/Attributes/SubscribeAttribute.cs:63]: Super nitpicky, but could use a newline between each method.

File: src/EvoSC.Common/Interfaces/Controllers/IController.cs [hunk 5]
```diff
@@ -5,9 +5,16 @@ public interface IController : IDisposable
     /// <summary>
     /// Set the controller's context object.
     /// </summary>
-    /// <param name="context"></param>
+    /// <param name="context">The context of the action that was executed.</param>
     public void SetContext(IControllerContext context);
+    /// <summary>
+    /// Get the action context of this controller instance.
+    /// </summary>
+    /// <returns></returns>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Interfaces/Controllers/IController.cs:13]: <returns>Controller context</returns>

File: src/EvoSC.Common/Interfaces/IEventManager.cs [hunk 6]
```diff
@@ -9,32 +10,52 @@ public interface IEventManager : IControllerActionRegistry
...
     public void Subscribe(Action<EventSubscriptionBuilder> builder);
+
+    /// <summary>
+    /// Subscribe to an event given the provided callback handler.
+    /// </summary>
+    /// <param name="name">The name of the event to subscribe to.</param>
+    /// <param name="handler">The callback method of the subscription.</param>
+    /// <param name="priority">Callback priority.</param>
+    /// <param name="runAsync">If true, the callback is run in a new thread.</param>
+    /// <typeparam name="TArgs"></typeparam>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Interfaces/IEventManager.cs:28]: Is a description of what TArgs is necessary here?

File: src/EvoSC.Common/Remote/GbxRemoteEvent.cs [hunk 7]
```diff
@@ -1,50 +1,63 @@
...
+public enum GbxRemoteEvent
 {
     /// <summary>
     /// When a player sends a message through in-game chat.
     /// </summary>
-    public const string PlayerChat = "GbxRemote.PlayerChat";
+    [EventIdentifier(Name = "GbxRemote.PlayerChat")]
+    PlayerChat,
     /// <summary>
     /// when a player successfully connects to the server.
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Remote/GbxRemoteEvent.cs:13]: When*, I know, I'm super nitpicky...

File: src/EvoSC.Common/Remote/GbxRemoteEvent.cs [hunk 8]
```diff
@@ -1,50 +1,63 @@
...
-    public const string ManialinkPageAnswer = "GbxRemote.ManialinkPageAnswer";
+    [EventIdentifier(Name = "GbxRemote.ManialinkPageAnswer")]
+    ManialinkPageAnswer,
     /// <summary>
     /// When the map list got modified.
     /// </summary>
-    public const string MapListModified = "GbxRemote.MapListModified";
+    [EventIdentifier(Name = "GbxRemote.MapListModified")]
+    MapListModified
 }
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Remote/GbxRemoteEvent.cs:63]: Excellent enum!

File: src/EvoSC.Modules/Interfaces/IModuleLoadContext.cs [hunk 9]
```diff
@@ -5,13 +5,38 @@
...
+    /// Unique ID of the module, used for referencing it at runtime. This ID is generated when the
+    /// module is loaded.
+    /// </summary>
     public Guid LoadId { get; init; }
+    /// <summary>
+    /// The type of the module's main class.
+    /// </summary>
     public Type? ModuleClass { get; init; }
+    /// <summary>
+    /// Meta info about a module-
```
Comment (AtomicLiquid) [inline src/EvoSC.Modules/Interfaces/IModuleLoadContext.cs:35]: module.*

File: src/EvoSC.Modules/Interfaces/IModuleManager.cs [hunk 10]
```diff
@@ -2,7 +2,15 @@
...
 
+/// <summary>
+/// Main manager for modules, provides loading/unloading and enabling/disabling of modules.
+/// </summary>
 public interface IModuleManager
 {
+    /// <summary>
+    /// Load all modules from a provided assembly.
+    /// </summary>
+    /// <param name="assembly">The assembly to search modules in.</param>
```
Comment (AtomicLiquid) [inline src/EvoSC.Modules/Interfaces/IModuleManager.cs:13]:
```suggestion
    /// <param name="assembly">The assembly which should be scanned for modules.</param>
```

File: src/EvoSC.Commands/Exceptions/CommandNotFoundException.cs [hunk 11]
```diff
@@ -1,5 +1,8 @@
 ﻿namespace EvoSC.Commands.Exceptions;
 
+/// <summary>
+/// Thrown when a command that does not exist was attempted retrieval.
```
[the remark this answers is not in the document]
Comment (snixtho) [reply src/EvoSC.Commands/Exceptions/CommandNotFoundException.cs:4]: Calling is not always the case.

File: src/EvoSC.Common/Config/ConfigServiceExtensions.cs (hunk 1 above)
Comment (snixtho) [reply src/EvoSC.Common/Config/ConfigServiceExtensions.cs:18]: This is an extension method, you can only call it on an object you already know about and what it is. I think documentation for the parameter that is extended is not neccessary.

File: src/EvoSC.Common/Config/Models/IDatabaseConfig.cs (hunk 2 above)
Comment (snixtho) [reply src/EvoSC.Common/Config/Models/IDatabaseConfig.cs:10]: only adding mysql as its the only one supported atm

**Review by AtomicLiquid — APPROVED** [review]

**Pull request merged**